client/src/views/workspaces/WorkspaceSharingModal.vue

```diff
@@ -276,7 +276,11 @@ const filteredUserRoles = computed(() => {
 const filteredSharedUserRoles = computed(() => {
   return filteredUserRoles.value
     .filter((userRole: UserRole) => userRole.role !== null)
-    .sort((item1, item2) => item1.user.humanHandle.label.localeCompare(item2.user.humanHandle.label));
+    .sort((item1, item2) => item1.user.humanHandle.label.localeCompare(item2.user.humanHandle.label))
```

Comment:
You're sorting the whole list twice, this is not the same as sorting by role first and by name second.